File: socketapp/src/aver.cpp
#include<iostream>
#include <stdlib.h>
#include<iostream>
#include <fstream>
using namespace std;

int main( int argc, const char* argv[] )
{
    if(argc!=4)
    {
        cerr<<"invalid number of arguments"<<endl;
        exit(1);
    }
    
    for(int i=0;i<argc;i++)
    {
      printf("%s\n",argv[i]);
    }
    
    ifstream txLog;
    
    long int start_time = atol(argv[1] );
    long int end_time = atol(argv[2] );
    const char* file=argv[3];
    
    txLog.open(file);
    
    int start_tx=20;
    int end_tx;    
    long int pre_time=start_time;
    
    double sum=0;
            long int time;
        int tx_old;
        int tx_new;

     printf("start time %ld\nend time %ld\n",start_time,end_time);
    while(txLog.good() )
    {


        txLog>>time;
        txLog>>tx_old;
        txLog>>tx_new;

        if(time<=start_time )
        {
            start_tx=tx_old;
            continue ;
        }
        else if(time>=end_time)
        {
            tx_new=tx_old;
            break ;
        }
        else
        {
            sum+=(time-pre_time)*tx_old;
            pre_time=time;
        }
    }
    
    txLog.close();
    sum+=(end_time-pre_time)*tx_new;
    
    double avg=sum/(end_time-start_time);
    printf("average power %f\n",avg);
}

File: socketapp/metrisis/aver.py
import sys
 
#start_time=1338562222
#end_time=1338564224
#filet ="tx.log.base"

start_time=int(sys.argv[1] )
end_time=int(sys.argv[2] )
filet=sys.argv[3]

f=open(filet,'r')

start_tx=20
end_tx=20
pre_time=start_time

sum=0
for line in f:
     time=int(line.split()[0] )
     tx_old=int( line.split()[1] )
     tx_new=int( line.split()[2] )
     
     #print time,  tx_old ,tx_new
     
     if time<=start_time :        
	  start_tx=tx_old;
     elif time>=end_time :
          tx_new=tx_old;
          break ;
     else :
	  sum+=(time-pre_time)*tx_old;
	  pre_time=time;
     
f.close();
sum+=(end_time-pre_time)*tx_new;
    
avg=sum/(end_time-start_time);
print "average power " ,avg
#print avg
     
     
     
     
     
     
     
     